Sources/PluginViewController/Configurations/SectionsConfiguration.swift

```diff
@@ -24,10 +27,50 @@ final class SectionsConfiguration: NSObject, Configuration {
         tableViewItems = sections.map { (section: NavigationPlugin) -> PluginItem in
             return makePluginItem(for: section)
         }
+        filteredTableViewItems = tableViewItems
+        flatPlugins = tableViewItems.flatMap { pluginItem in
+            collectPlugins(in: pluginItem.plugin, sectionTitle: pluginItem.title)
+        }
+    }
+
+    func filterData(with text: String) {
+        defer {
+            tableView?.reloadData()
+        }
+        guard text.isEmpty == false else {
+            filteredTableViewItems = tableViewItems
+            return
+        }
+
+        let filteredFlatPlugins = flatPlugins.filter { plugin in
+            plugin.keywords.lowercased().contains(text)
+        }
+        let resultPluginItems = filteredFlatPlugins.map { plugin -> PluginItem in
+            PluginItem(title: plugin.title.string, plugin: plugin, children: [])
+        }
```

Comment:
I think, here you can do it in one loop

Reply:
Yes, for example, use compactMap and return nil for unfiltered items